x/mongo/driver/topology/topology.go

```diff
@@ -542,10 +542,19 @@ func (t *Topology) selectServerFromDescription(desc description.Topology,
 		return desc.Servers, nil
 	}
 
-	var allowed []description.Server
-	for _, s := range desc.Servers {
+	allowedIndexes := make([]int, 0, len(desc.Servers))
+	for i, s := range desc.Servers {
 		if s.Kind != description.Unknown {
-			allowed = append(allowed, s)
+			allowedIndexes = append(allowedIndexes, i)
```

Comment:
This would be more uniform and cheaper solution:

```go
        allowedIndexes := make([]int,  len(desc.Servers))
	for i, s := range desc.Servers {
		if s.Kind != description.Unknown {
			allowedIndexes[i] = i
		}
	}
```

```
❯ go test -bench=BenchmarkSelectServerFromDescription
goos: darwin
goarch: arm64
pkg: go.mongodb.org/mongo-driver/x/mongo/driver/topology
BenchmarkSelectServerFromDescription/AllFit-10           3274647               363.8 ns/op           896 B/op          1 allocs/op
BenchmarkSelectServerFromDescription/AllButOneFit-10     3274470               364.3 ns/op           896 B/op          1 allocs/op
BenchmarkSelectServerFromDescription/HalfFit-10          3292062               365.5 ns/op           896 B/op          1 allocs/op
BenchmarkSelectServerFromDescription/OneFit-10           3220062               368.9 ns/op           896 B/op          1 allocs/op
```